### PR TITLE
Add vcomlink test

### PR DESCRIPTION
**Pull request by shjala**
This is a test for https://github.com/lf-edge/eve/pull/4142, but I don't know which one should go first ?!

- [x] Wait for next release with https://github.com/lf-edge/eve/pull/4204
- [x] Open a PR to set eve tag to the latest
- [x] Rebase against latest master
- [x] ~Reduce the test verbosity before merging~

### CODE REVIEW

File: tests/vcom/base.go
Comment (milan-zededa) [inline tests/vcom/base.go:42]: I see that this initialization repeats in every Go tests.
How about to create a framework for eden Go tests, which would handle all this test initialization, minimize boiler-plate and provide various helper methods (what you currently have in pkg/remote)?

File: pkg/remote/utils.go
Comment (milan-zededa) [inline pkg/remote/utils.go:1]: Note that "remote" is not an ideal name. In eden we use "remote" to denote physical edge node controlled by eden over the network - as opposed to locally started VM running EVE. However, this package I suspect is mostly going to be used for testing EVE running in a VM.

File: pkg/remote/utils.go
Comment (milan-zededa) [inline pkg/remote/utils.go:225]: Which app? Why is this method not taking appName as an argument?

File: pkg/remote/utils.go
Comment (milan-zededa) [inline pkg/remote/utils.go:257]: Same question as asked for AppSSHExec.

**Comment by milan-zededa** [comment]
> This is a test for https://github.com/lf-edge/eve/pull/4142, but I don't know which one should go first ?!

If this test depends on the EVE PR and would fail until it is merged and released, then this PR should wait.
I typically mark such eden PR as draft until there is EVE version with the tested functionality available. Then I update `DefaultEVETag` in `pkg/defaults` as part of the PR and mark it as ready for review.

**Comment by shjala** [comment]
> If this test depends on the EVE PR and would fail until it is merged and released, then this PR should wait.

Yeah, I get it. But it is a conundrum, how I can show that my PR in eve works fine without merging this test, and on the other hand merging this breaks the eden tests.

File: pkg/remote/utils.go
Comment (shjala) [reply pkg/remote/utils.go:225]: I was operating on the fact that my test only needs one app, but right I will change it to accept app name.

File: pkg/remote/utils.go
Comment (shjala) [reply pkg/remote/utils.go:1]: yes, this is more of auxiliary package for testing stuff on EVE and VMs, what do you think would be a good name?

File: tests/vcom/base.go
Comment (shjala) [reply tests/vcom/base.go:42]: sure, later I will refactor the test/sec and tests/aziot after this one gets merged.

File: tests/vcom/vcom_test.go
Comment (milan-zededa) [inline tests/vcom/vcom_test.go:47]: This will likely print out `vcomlink is not running <nil>` when there is a problem with vsock. I think that printing `stat` instead of (or additionally to) `err` will be more useful.

Comment (shjala) [reply tests/vcom/vcom_test.go:47]: got catch!

**Comment by shjala** [comment]
This is ready to be merge as soon as we get the new eve release.

**Comment by uncleDecart** [comment]
@shjala so EVE 13.3.0 supports the test?

**Comment by shjala** [comment]
> @shjala so EVE 13.3.0 supports the test?

yes, it runs green locally, lets see what happens here.

**Comment by shjala** [comment]
This is really frustrating, it runs fine locally (running smoke test) then fails to wait for the app to be ready, I increased the timer to 30m but still...

**Comment by uncleDecart** [comment]
We need to think if there's something in Eden or problem in runners...

**Comment by giggsoff** [comment]
> This is really frustrating, it runs fine locally (running smoke test) but here fails to wait for the app to be ready, I increased the timer to 30m but still...

Inside attached artifacts (in console logs) I can see `ERROR: App vcomlink-sleepy_leakey uuid 53ff1865-61bf-43dd-bab4-ab7c1ffe2fb6 state INSTALLED error: failed to run domain 53ff1865-61bf-43dd-bab4-ab7c1ffe2fb6.1.1: not based on an OCI image`. AFAIK it indicates that you try to run VM (based on qcow2 image) on host without hardware assistance acceleration (or with app in no-hyper mode in config).

BTW it would be nice to show timestanps and errors if any in state transition in test logs (from `AppWaitForRunningState` function). It will helps to detect such problems quickly without diving into artifacts.

**Comment by shjala** [comment]
> Inside attached artifacts (in console logs) I can see `ERROR: App vcomlink-sleepy_leakey uuid 53ff1865-61bf-43dd-bab4-ab7c1ffe2fb6 state INSTALLED error: failed to run domain 53ff1865-61bf-43dd-bab4-ab7c1ffe2fb6.1.1: not based on an OCI image`. AFAIK it indicates that you try to run VM (based on qcow2 image) on host without hardware assistance acceleration (or with app in no-hyper mode in config).

I see the eden default DefaultEVEH is KVM and in the app config NoHyper is set to false, are you suggesting I should move this to virtualization test suite? 

> BTW it would be nice to show timestanps and errors if any in state transition in test logs (from `AppWaitForRunningState` function). It will helps to detect such problems quickly without diving into artifacts.

Good point, I will add it.

**Comment by shjala** [comment]
I see `require_virtualization: true` only for virtualization test in github actions, I move this test to virt.

**Comment by shjala** [comment]
@giggsoff Thanks for your help.
it is now passing as part of virtualization tests: 
```
--- PASS: TestEdenScripts (0.00s)
    --- PASS: TestEdenScripts/eden_vcom (119.12s)
```

@milan-zededa @uncleDecart some other tests are failing but I'm not touching any part relevant to those, general flakiness? anyways this PR is ready.

**Review by uncleDecart — APPROVED** [review]
LGTM from my side, if @milan-zededa is okay with patch I'd merge it

**Review by milan-zededa — APPROVED** [review]

**Pull request merged**
